src/Workspaces/Core/Portable/PublicAPI.Unshipped.txt

```diff
@@ -1,12 +1,21 @@
+Microsoft.CodeAnalysis.ApplyChangesKind.AddSolutionAnalyzerReference = 20 -> Microsoft.CodeAnalysis.ApplyChangesKind
+Microsoft.CodeAnalysis.ApplyChangesKind.RemoveSolutionAnalyzerReference = 21 -> Microsoft.CodeAnalysis.ApplyChangesKind
 Microsoft.CodeAnalysis.Editing.DeclarationModifiers.IsExtern.get -> bool
 Microsoft.CodeAnalysis.Editing.DeclarationModifiers.IsVolatile.get -> bool
 Microsoft.CodeAnalysis.Editing.DeclarationModifiers.WithIsExtern(bool isExtern) -> Microsoft.CodeAnalysis.Editing.DeclarationModifiers
 Microsoft.CodeAnalysis.Editing.DeclarationModifiers.WithIsVolatile(bool isVolatile) -> Microsoft.CodeAnalysis.Editing.DeclarationModifiers
 Microsoft.CodeAnalysis.Editing.SyntaxGenerator.ElementBindingExpression(params Microsoft.CodeAnalysis.SyntaxNode[] arguments) -> Microsoft.CodeAnalysis.SyntaxNode
 Microsoft.CodeAnalysis.Project.RemoveDocuments(System.Collections.Immutable.ImmutableArray<Microsoft.CodeAnalysis.DocumentId> documentIds) -> Microsoft.CodeAnalysis.Project
+Microsoft.CodeAnalysis.Solution.AddAnalyzerReference(Microsoft.CodeAnalysis.Diagnostics.AnalyzerReference analyzerReference) -> Microsoft.CodeAnalysis.Solution
```

Comment:
This is simpler for the host. Otherwise, the host would need to 
- remember somewhere separate from the solution what those references are
- add the references when projects are added
- be careful to include the references when calling project.WithAnalyzerReferences

Also, these references are treated differently by the diagnostic incremental analyzer. It may or may not be an artifact of the previous approach, but I can imagine some caching can be done differently when you know the references are shared by all projects.

Reply:
> consumers having to learn about these new concepts and potentially requiring updates.

What consumers do you have on mind? The only consumer I know of is the diagnostic incremental analyze. There can't be any existing consumers that wouldn't know about these references (if they needed them) since they were not available from the solution at all previously. Going forward, I don't think it's hard to learn that one needs to concat solution.AnalyzerReferences + project.AnalyzerReferences.

Reply:
Another difference - project analyzer references are displayed in UI per project. The host references aren't and I don't think we would want them to be listed in each project. If anything they should be listed under dedicated solution node.

Reply:
I was thinking of code like [this icky thing](https://github.com/dotnet/roslyn/blob/01e408117b713b796ab51f5cf20b0bd32524cac4/src/VisualStudio/Core/Def/Implementation/AnalyzerDependency/AnalyzerDependencyCheckingService.cs#L80-L85) where we are collecting analyzers and then looking for issues. But then I guess the design actually keeps that code working as it does today. I'm convinced.